torchdata/datapipes/iter/transform/bucketbatcher.py

```diff
@@ -37,15 +39,61 @@ class InBatchShufflerIterDataPipe(IterDataPipe[DataChunk[T_co]]):
 
     def __init__(self, datapipe: IterDataPipe[DataChunk[T_co]]):
         self.datapipe = datapipe
+        self._enabled = True
+        self._seed: Optional[int] = None
+        self._rng = random.Random()
+
+    def set_shuffle(self, shuffle=True):
+        self._enabled = shuffle
+        return self
+
+    def set_seed(self, seed: int):
+        self._seed = seed
+        return self
 
     def __iter__(self) -> Iterator[DataChunk[T_co]]:
-        for batch in self.datapipe:
-            random.shuffle(batch)
-            yield batch
+        if not self._enabled:
+            for batch in self.datapipe:
+                yield batch
+        else:
+            for batch in self.datapipe:
+                new_batch = self._rng.sample(batch, len(batch))
+                yield DataChunk(new_batch)
+
+    def reset(self) -> None:
+        if self._enabled and self._seed is None:
+            self._seed = int(torch.empty((), dtype=torch.int64).random_().item())
+        self._rng.seed(self._seed)
+        self._seed = None
```

Comment:
No strong opinion about this feel free to ignore:
Should we skip over these two lines as well if `self._enabled == False`?

The practical difference is that `self._rng` and `self._seed` will not change unless `self._enabled == True`.
Tbh I doubt we will run into a case like this but anyway

Reply:
Actually `shuffler` is already doing that so we can keep it that way.

Reply:
This is actually a good catch considering a super corner case that `self._enabled = False` and `self._seed is not None` and `seed` will be eliminated by this `reset` function.

Thank you. But, I might open a separate PR to change both of them.

Reply:
> But, I might open a separate PR to change both of them.

Forget about it. I will change it right away